src/test/java/net/snowflake/client/jdbc/ResultSetVectorLatestIT.java

```diff
@@ -0,0 +1,117 @@
+package net.snowflake.client.jdbc;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.List;
+import net.snowflake.client.category.TestCategoryResultSet;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * ResultSet integration tests for the latest JDBC driver. This doesn't work for the oldest
+ * supported driver. Revisit this tests whenever bumping up the oldest supported driver to examine
+ * if the tests still is not applicable. If it is applicable, move tests to ResultSetIT so that both
+ * the latest and oldest supported driver run the tests.
+ */
+@Category(TestCategoryResultSet.class)
+@RunWith(Parameterized.class)
+public class ResultSetVectorLatestIT extends ResultSet0IT {
+
+  private final String queryResultFormat;
+
+  public ResultSetVectorLatestIT(String queryResultFormat) {
+    super(queryResultFormat);
+    this.queryResultFormat = queryResultFormat;
+  }
+
+  @Parameterized.Parameters(name = "format={0}")
+  public static List<String> queryResultFormats() {
+    return Arrays.asList("json", "arrow");
+  }
+
+  @Test
+  public void testGetIntVectorAsIntArray() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      enforceQueryResultFormat(stmt);
+      Integer[] vector = {-1, 5};
+      ResultSet resultSet = stmt.executeQuery("select " + vectorToString(vector, "int"));
+      assertTrue(resultSet.next());
+      Integer[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Integer.class);
+      assertEquals(vector, result);
+    }
+  }
+
+  @Test
+  public void testGetIntVectorAsLongArray() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      enforceQueryResultFormat(stmt);
+      Long[] vector = {-1L, 5L};
+      ResultSet resultSet = stmt.executeQuery("select " + vectorToString(vector, "int"));
```

Comment:
should it be int or bigint?

Reply:
it should be int. Vectors can be of type of int or float only. This tests shows that we can easyli cast vector of ints to array of longs.